### ISSUE TITLE
4.0.0: Cast dialogue list showing up empty on iOS 14.3 [iOS]

### ISSUE THREAD

**Issue by maliotti** [react-native-google-cast/react-native-google-cast#276]
For version `4.0.0`, when showing the cast dialogue on iOS 14.3, the cast device list is empty on both the simulator and a physical device. Works fine on iOS 13.4. Any thoughts on how to get this working again?

As a side note, I noticed that version `4.0.0` has a require cycle warning:
`Require cycle: node_modules/react-native-google-cast/src/index.ts -> node_modules/react-native-google-cast/src/api/CastContext.ts -> node_modules/react-native-google-cast/src/api/SessionManager.ts -> node_modules/react-native-google-cast/src/api/CastSession.ts -> node_modules/react-native-google-cast/src/api/RemoteMediaClient.ts -> node_modules/react-native-google-cast/src/index.ts`

![ios-missing-cast-destinations](https://user-images.githubusercontent.com/5532759/104968044-b36c0000-5999-11eb-94d7-5b900a14ad0e.PNG)

**Comment by rmccue** [comment]
@maliotti Have you granted the necessary network permissions? Check the debug log in XCode to see if the Google Cast SDK is complaining about it.

Note that while the setup guide says "If using the default receiver, you may delete that line." for the NSBonjourServices line, I found I had to add the default receiver's Bonjour service ID (`_CC1AD845._googlecast._tcp`).

(The devices list works for me on both simulator and device in 14.3, FWIW.)

**Comment by official-bharat** [comment]
#279 Please help me to fix this issue
@maliotti

**Comment by maliotti** [comment]
Hi, @rmccue! Thanks for sharing this info! I was indeed getting a warning in the console about not having the service ID in there. I added that to the `Info.plist` file and the warning went away. However, that was only part of my issue. In addition to the missing service ID, I was not using the CastButton to display the cast dialogue, so the device discovery wasn't firing. I set `options.startDiscoveryAfterFirstTapOnCastButton = false` and that did the trick! Now the cast devices show up and everything seems to work as expected. Thank you again for your help with this. I truly appreciate it!
